mat/cholesky.go

```diff
@@ -340,6 +340,70 @@ func (c *Cholesky) Scale(f float64, orig *Cholesky) {
 	c.cond = orig.cond // Scaling by a positive constant does not change the condition number.
 }
 
+// AppendOneSym computes the Cholesky decomposition of the original matrix A
+// with the data appended to the original matrix A, storing the result into the
+// receiver. That is orig contains the Cholesky decomposition of A, and at completion
+// it contains the Cholesky decomposition of
+//  [A a]
+//  [a k]
+// where k = r[len(r)-1] and a = r[:len(r)-1]. In order for the updated matrix
+// to be positive definite, it must be the case that n > a' A^-1 a. If this
+// condition does not hold then AppendOneSym will return false and the receiver
+// will not be updated.
+//
+// AppendRow will panic if len(r) != orig.Size()+1 or if orig does not contain
+// a valid decomposition.
+func (chol *Cholesky) AppendOneSym(orig *Cholesky, r Vector) (ok bool) {
+	n := orig.Size()
+	if r.Len() != n+1 {
+		panic(badSliceLength)
+	}
+	if !orig.valid() {
+		panic(badCholesky)
+	}
+
+	// The algorithm is commented here, but see also
+	//  https://math.stackexchange.com/questions/955874/cholesky-factor-when-adding-a-row-and-column-to-already-factorized-matrix
+	// We have A and want to compute the Cholesky of
+	//  [A a]
+	//  [a k]
+	// We want
+	//  [U c]
+	//  [0 d]
+	// to be the updated Choleksy, and so it must be that
+	//  [A a] = [U' 0 ] [U c]
+	//  [a k]   [c' d'] [0 d]
+	// Thus, we need
+	//  1) A = U'U (true by orig being a valid Cholesky decomposition),
+	//  2) U' * c = a  =>  c = U'^-1 a
+	//  3) c'*c + d'*d = n  =>  d = sqrt(n-c'*c)
+
+	// First, compute c = U'^-1 a
+	a := NewVecDense(n, nil)
+	for i := 0; i < n; i++ {
```

Comment:
I'm surprised that `CopyVec` takes a `mat.VecDense`. Ah, it's one of the unfinished items in #167. Please add a TODO to change this to `a.CopyVec(...)` when that is fixed.

Reply:
Done.